### PR TITLE
Add device detection hostcall

### PR DESCRIPTION
**Pull request by silentbicycle**
Add hostcall for device detection, closely patterned on geolocation.

The test that executes the hostcall needs a new SDK, so a second commit comments them out for now. The tests pass locally when built with the new SDK.

### CODE REVIEW

File: lib/src/config/unit_tests.rs [hunk 1]
```diff
@@ -847,6 +847,38 @@ mod inline_toml_dictionary_config_tests {
...
+    use {
+        super::read_local_server_config,
+        crate::error::{
+            DeviceDetectionConfigError, FastlyConfigError::InvalidDeviceDetectionDefinition,
+        },
+    };
+
+    /// Check that Device Detection definitions have a valid `format`.
+    #[test]
+    fn geolocation_has_a_valid_format() {
```
Comment (acfoltzer) [inline lib/src/config/unit_tests.rs:864]:
```suggestion
    fn device_detection_has_a_valid_format() {
```

File: test-fixtures/src/bin/grpc.rs [hunk 2]
```diff
@@ -1,13 +1,13 @@
-use fastly::{Backend, Error, Request};
```
Comment (acfoltzer) [inline test-fixtures/src/bin/grpc.rs:1]: Huh, the formatting step in CI must be missing these if rustfmt changed these despite previous successful commits. Something to worry about later

**Review by acfoltzer — CHANGES_REQUESTED** [review]
Just a small copypaste fix, then ready to go

File: lib/src/config/unit_tests.rs (hunk 1 above)
Comment (silentbicycle) [reply lib/src/config/unit_tests.rs]: Fixed in c6ea312.

**Review by acfoltzer — APPROVED** [review]
thank you!

**Pull request merged**